scipy/stats/_continuous_distns.py

```diff
@@ -8572,6 +8572,111 @@ def _rvs_scalar(self, a, b, numsamples=None, random_state=None):
 truncnorm = truncnorm_gen(name='truncnorm', momtype=1)
 
 
+class truncpareto_gen(rv_continuous):
+    r"""An upper truncated Pareto continuous random variable.
+
+    %(before_notes)s
+
+    See Also
+    --------
+    pareto
+
+    Notes
+    -----
+    The probability density function for `truncpareto` is:
+
+    .. math::
+
+        f(x, b, c) = \frac{b}{1 - c^{-b}} \frac{1}{x^{b+1}}
+
+    for :math:`b > 0`, :math:`c > 1` and :math:`1 \le x \le c`.
+
+    `truncpareto` takes :math:`b`, :math:`c` as shape parameters.
+
+    Notice that the upper truncation value :math:`c` is defined in
+    standardized form:
+
+    .. math::
+
+        1 = (u_l - loc)/scale \quad ; \quad c = (u_r - loc)/scale
+
+    where :math:`u_l` and :math:`u_r` are the specific left and right
+    truncation values, respectively. In other words, the support of the
+    distribution becomes :math:`(scale + loc) <= x <= (c*scale + loc)` when
+    :math:`loc` and/or :math:`scale` are provided.
+
+    %(after_notes)s
+
+    References
+    ----------
+    .. [1] Burroughs, S. M., and Tebbens S. F.
+        "Upper-truncated power laws in natural systems."
+        Pure and Applied Geophysics 158.4 (2001): 741-757.
```

Comment:
This reference doesn't appear to have the formulas for all of the overridden methods below. Did you derive those yourself? If so, could you provide your notes? If not, could you point to a specific reference with each? As a reviewer, I would check all of the methods against their source, so whatever you can do to facilitate that process would be appreciated.

Reply:
To be honest, I mostly put this reference for context/example of application.
I did most of the derivations; it was easier to accommodate SciPy's framework this way.
It was a bit of a scribbly mess, so I took the time to clean it up! All the overridden methods, except the _log[pdf|cdf|sf] (really straightforward to derive) should be covered in the attached file :)

[notes.pdf](https://github.com/scipy/scipy/files/8832762/notes.pdf)